### DISCUSSION TITLE
Support CNAV ephemeris types

### DISCUSSION THREAD

**Discussion by ljbade** [nav-solutions/rinex#76] [Ideas]
With RINEX 4 it is now possible to get broadcast ephemerides for the newer signals such as GPS CNAV or BDS CNAV. However it seems the current implementation only recognises the ephemeris types from RINEX 3 such as LNAV.

I am interested in computing the position difference between the different ephemeris types for the same satellite so it would be useful to add support for:
- GPS CNAV and CNAV-2
- QZSS CNAV and CNAV-2
- BDS CNAV-1, CNAV-2 and CNAV-3

The recent IGS broadcast navigation files have examples of most of these new formats. An example is https://cddis.nasa.gov/archive/gnss/data/daily/2023/brdc/BRD400DLR_S_20230750000_01D_MN.rnx.gz

The IGS file doesn't seem to yet have an example of GPS CNAV-2 yet, which is from the GPS L1C signal.

**Comment by larsnaesbye** [comment]
@ljbade I think I'll make this into an issue.

**Comment by ljbade** [comment]
@larsnaesbye thanks. I did find some other issues, and I been working on a few hacky fixes to get things working, but I would like to get these fixed properly. I'm happy to open some issues/PRs for what I have so far.
